File: test/cs/case27_class_instance.cs
using System.Collections.Generic;
using System.IO;
using System;

public static class Program
{
    public class Box100
    {
        public int seed;

        public Box100(int seed)
        {
            this.seed = seed;
        }
        public int next()
        {
            this.seed = (this.seed + 1);
            return this.seed;
        }
    }

    public static void Main(string[] args)
    {
        Box100 b = new Box100(3);
        Pytra.CsModule.py_runtime.print(b.next());
    }
}

File: test/rs/case27_class_instance.rs
#[path = "../../src/rs_module/py_runtime.rs"]
mod py_runtime;
use py_runtime::{math_cos, math_exp, math_sin, math_sqrt, perf_counter, py_bool, py_grayscale_palette, py_in, py_isalpha, py_isdigit, py_len, py_print, py_save_gif, py_slice, py_write_rgb_png};

// このファイルは自動生成です（native Rust mode）。

#[derive(Clone)]
struct Box100 {
    seed: i64,
}

impl Box100 {
    fn new(seed: i64) -> Self {
        let mut self_obj = Self {
            seed: 0,
        };
        self_obj.seed = seed;
        self_obj
    }

    fn next(&mut self) -> i64 {
        self.seed = self.seed + 1;
        return self.seed;
    }
}

fn main() {
    let mut b: Box100 = Box100::new(3);
    py_print(b.next());
}
